astropy/modeling/tests/test_parameters.py

```diff
@@ -603,9 +603,10 @@ def test_validator(self):
         param = Parameter(name="test", default=[1, 2, 3, 4])
         assert param._validator is None
 
-        valid = mk.MagicMock()
-        param.validator(valid)
-        assert param._validator == valid
+        # valid = mk.MagicMock()
+        # MESSAGE = r"The validator property of the Parameter class is deprecated.*"
+        # with pytest.raises(AstropyDeprecationWarning, match=MESSAGE):
+        #     param.validator(valid)
```

Comment:
Is this the matching test change for the warning? 
```suggestion
        # valid = mk.MagicMock()
        # MESSAGE = r"The validator property of the Parameter class is deprecated.*"
        # with pytest.warns(AstropyDeprecationWarning, match=MESSAGE):
        #     param.validator(valid)
```
If you do want to include the warning, this change will catch it, `raises` only catches exceptions.